src/app/core/services/navigation/nav-tree.service.ts

```diff
@@ -76,9 +76,12 @@ export abstract class NavTreeService<ContentT extends RoutedContentInfo> {
     shareReplay(1),
   );
 
+  private navigationRootElementId$ = new BehaviorSubject<string | undefined>(undefined);
   navigationNeighbors$: Observable<FetchState<NavigationNeighbors|undefined>> = this.state$.pipe(
+    repeatLatestWhen(this.navigationRootElementId$), // refresh when navigation root element changes
     mapStateData(navData => {
-      if (!navData.selectedElementId) return undefined;
+      const navigationRootElementId = this.navigationRootElementId$.value;
```

Comment:
`Subject.value` should only be used in specific case when you exit Rx.
Here, you have cleaner way to combine the subject.

(alternative: you use regular variable and not a subject... anyway it is not supposed to change once the app is loaded)

Reply:
fixed